syntax = "proto3";

package inference;

// Request carries feature name -> value
message PredictRequest {
    map<string, float> features = 1;
    string model_name = 2; // optional: in case of multiple models
}

// Response carries prediction value
message PredictResponse {
    // repeated because multiple predictions can be returned
    repeated float prediction = 1;
    string model_version = 2; 
}

// gRPC service definition
service InferenceService {
    // Prediction endpoint
    rpc Predict (PredictRequest) returns (PredictResponse);
}
